Cargo.toml

```diff
@@ -7,109 +7,35 @@ edition = "2018"
 [dependencies]
 hex = { version = "0.4.3", default-features = false, features = ["alloc"] }
 log = { version = "0.4", optional = true }
-primitive-types = { version = "0.6.1", optional = true, features = ["codec"] }
+primitive-types = { version = "0.10.1", optional = true, features = ["codec"] }
 serde = { version = "1.0", optional = true, features = ["derive"] }
 serde_json = { version = "1.0", optional = true }
 thiserror = { version = "1.0", optional = true }
 ws = { version = "0.9.1", optional = true, features = ["ssl"] }
+codec = { package = 'parity-scale-codec', version = "2.0.0", default-features = false,  features = ['derive']}
+
+# Substrate dependencies
+sp-core = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master", features = ["full_crypto"] }
+metadata = { version = "14.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master", package = "frame-metadata" }
+sp-version = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "master", package = "sp-version", optional = true }
+balances = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "master", package = "pallet-balances", optional = true }
+staking = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "master", package = "pallet-staking", optional = true }
+system = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "master", package = "frame-system", optional = true }
+transaction-payment = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master", package = "pallet-transaction-payment", optional = true }
+sp-rpc = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master", optional = true }
+sp-runtime = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master" }
+support = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", package = "frame-support", branch = "master" }
+sp-std = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master" }
+# need to add this for the app_crypto macro
+sp-application-crypto = { version = "4.0.0-dev", default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master", features = ["full_crypto"] }
+
```

Comment:
Oh again, so nice. :).

For me, this file is a good example that the one-liner is a better option.